internal/server/middleware.go

```diff
@@ -162,15 +162,15 @@ func requireAccessKey(c *gin.Context, db *data.Transaction, srv *Server) (access
 	accessKey, err := data.ValidateRequestAccessKey(db, bearer)
 	if err != nil {
 		if errors.Is(err, data.ErrAccessKeyExpired) {
-			return u, err
+			return u, AuthenticationError{Message: "access key has expired"}
 		}
 		return u, fmt.Errorf("%w: invalid token: %s", internal.ErrUnauthorized, err)
 	}
 
 	if accessKey.Scopes.Includes(models.ScopePasswordReset) {
 		// PUT /api/users/:id only
 		if c.Request.URL.Path != "/api/users/"+accessKey.IssuedFor.String() || c.Request.Method != http.MethodPut {
-			return u, fmt.Errorf("%w: temporary passwords can only be used to set new passwords", internal.ErrUnauthorized)
+			return u, fmt.Errorf("%w: temporary passwords can only be used to set new passwords", access.ErrNotAuthorized)
```

Comment:
Changed to `403` (not authorized) because the user has authenticated successfully, but their access key only gives them permission for one operation.